### PR TITLE
Travis generated bundles

### PR DESCRIPTION
**Pull request by hypest**
This PR adds an extra job in Travis which produces the JS bundle and uploads them to Amazon S3.

The final URLs are:
`https://s3-us-west-1.amazonaws.com/gutenberg-mobile-js-bundle/wordpress-mobile/gutenberg-mobile/<commit hash>/android/App.js`

`https://s3-us-west-1.amazonaws.com/gutenberg-mobile-js-bundle/wordpress-mobile/gutenberg-mobile/<commit hash>/android/App.js.map`

`https://s3-us-west-1.amazonaws.com/gutenberg-mobile-js-bundle/wordpress-mobile/gutenberg-mobile/<commit hash>/ios/App.js`

`https://s3-us-west-1.amazonaws.com/gutenberg-mobile-js-bundle/wordpress-mobile/gutenberg-mobile/<commit hash>/ios/App.js.map`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+  - secure: QdhoQVjffAiAPcUjN4oPUPuWJX3w7wJQS+p1VE3HWjMGWEFxlEd4k1F91L7uTcibbaogSCZLmh6gbLlcykS9Hi7rD+eiarAlH7acitpuOtTOctrqmx3TmaHV+2Z5aBj33vhA7Ug7/Rmi6JJa83z1jT1AYz1bggDE+ybUp2vQB9NWcuvXBCf6gFnVrYmiCUeJsCQmZubJCvDxxXjNhw9QfpVDCnPXilBQtzKx5qWIx+Ir8kOxTP8xucaqi6VD+rZtT3ifSEZ4suWS68+ZI9AreDAhIaArI5Pmdv/KWaum2LVfeIvxt6lMNOUpoj+83yilapPSoQS6cgOtEmdr7U8ZMVdueA9Nada96BjRErrGNYMcAglMHrNpFfYUVj8CdqlN9IlAQyr7rMz9xWKweCexFTv7bOB6rr38bscZBbhDzo9Iswlr/3UnYeV8VoVXevSqK1+eLFJdvAMR04hT4wa2g1vsujbO/SX2MoEeEiGkOG3Utb5x4d/iHoSQ0vy17vNZ1KCciUpKlWqh8MxkXwWGFYUNilEr9P0SeOU0Qt/8DRzzTQ0p9+j4ihlpDXFk2nv9HBMdQdVpoNqhPyNH2uxSIkuzizrWQtId13MtrdlFEVdpiTjr9OVMra+KuHlUfNtuPU745xhES/SpcLrDH1RZWiBdBPaLPg6r7UFKem1q3oM=
```
Comment (jtreanor) [inline .travis.yml]: I'm curious; why are there three secrets here?

Comment (hypest) [reply .travis.yml]: Hey James, the third one is `ARTIFACTS_BUCKET` but, that's not really needed being encrypted so, moved it to the normal `env` section of the job. Done with 95abaee.

Let me know if you're interested in the other two too. Thanks!

File: package.json [hunk 2]
```diff
@@ -60,8 +60,8 @@
...
     "bundle": "yarn bundle:android && yarn bundle:ios",
-    "bundle:android": "mkdir -p bundle/android && yarn react-native bundle --platform android --dev false --entry-file index.js --assets-dest bundle/android --bundle-output bundle/android/App.js --sourcemap-output bundle/android/App.js.map",
-    "bundle:ios": "mkdir -p bundle/ios && yarn react-native bundle --platform ios --dev false --entry-file index.js --assets-dest bundle/ios --bundle-output bundle/ios/App.js --sourcemap-output bundle/ios/App.js.map",
+    "bundle:android": "GITHASH=$(git rev-parse HEAD) && mkdir -p bundle/$GITHASH/android && yarn react-native bundle --platform android --dev false --entry-file index.js --assets-dest bundle/$GITHASH/android --bundle-output bundle/$GITHASH/android/App.js --sourcemap-output bundle/$GITHASH/android/App.js.map",
+    "bundle:ios": "GITHASH=$(git rev-parse HEAD) && mkdir -p bundle/$GITHASH/ios && yarn react-native bundle --platform ios --dev false --entry-file index.js --assets-dest bundle/$GITHASH/ios --bundle-output bundle/$GITHASH/ios/App.js --sourcemap-output bundle/$GITHASH/ios/App.js.map",
```
[the remark this answers is not in the document]
Comment (hypest) [reply package.json]: Hey Koke, took another approach (01505c4) by using the `TRAVIS_COMMIT` predefined Travis environment variable and using it directly in the S3 target path. This way there is no need to alter the current `bundle` directly structure or npm script at all. Let me know what you think.

This is based on the `TRAVIS_COMMIT` idea you shared over Slack so, thanks for that!

**Comment by hypest** [comment]
Thanks for the review Koke and James!

**Pull request merged**